lib/sla.js

```diff
@@ -125,6 +127,9 @@ function originateChannel(client, inbound, bridge, trunkData) {
       var hangup = Q.denodeify(unanswered.channel.hangup.bind(
           unanswered.channel));
       hangup();
+      var deviceState = Q.denodeify(client.deviceStates.update.bind(client));
```

Comment:
This pattern of getting the device state from the client, setting up the device to change, and changing its state, is pretty common throughout this patch.

I'd write a small function that does just that, taking in the name of the device to change and the state to set it to.